back/scripts/enrichment/communities_enricher.py

```diff
@@ -16,10 +19,49 @@ def get_dataset_name(cls) -> str:
 
     @classmethod
     def get_input_paths(cls, main_config: dict) -> typing.List[Path]:
-        return [CommunitiesSelector.get_output_path(main_config)]
+        return [
+            CommunitiesSelector.get_output_path(main_config),
+            BaremeEnricher.get_output_path(main_config),
+        ]
 
     @classmethod
     def _clean_and_enrich(cls, inputs: typing.List[pl.DataFrame]) -> pl.DataFrame:
-        communities, *_ = inputs
-        # Data analysts, please add your code here!
-        return communities
+        communities, bareme = inputs
+
+        current_year = datetime.now().year
+        target_year = current_year - 2
+        return (
+            communities.join(
+                bareme.filter(pl.col("annee") == target_year).select(
+                    ["siren", "mp_score", "subventions_score"]
+                ),
+                on="siren",
+                how="left",
+            )
+            .with_columns(
+                [
+                    cls._map_score_to_numeric("mp_score").alias("mp_score_num"),
+                    cls._map_score_to_numeric("subventions_score").alias("sub_score_num"),
+                ]
+            )
+            .with_columns(
+                [
+                    ((pl.col("mp_score_num") + pl.col("sub_score_num")) / 2)
+                    .floor()
+                    .cast(pl.Int64)
+                    .alias("global_score_num"),
+                ]
+            )
+            .with_columns([cls._map_numeric_to_score("global_score_num").alias("global_score")])
+            .drop(["mp_score_num", "sub_score_num", "global_score_num"])
```

Comment:
```suggestion
           .with_columns([
    # Mapping des scores lettres vers numériques
    CommunitiesEnricher._map_score_to_numeric("mp_score").alias("mp_score_num"),
    CommunitiesEnricher._map_score_to_numeric("subventions_score").alias("sub_score_num"),

    # Calcul du score global (moyenne)
    ((pl.col("mp_score_num") + pl.col("sub_score_num")) / 2)
        .floor()
        .cast(pl.Int64)
        .alias("global_score_num"),

    
    CommunitiesEnricher._map_numeric_to_score("global_score_num").alias("global_score")
])
.drop(["mp_score_num", "sub_score_num", "global_score_num"])
```

Je propose un seul with columns et une séparation "logique" des transformations: Mapping puis Compute puis Restitution